CHANGELOG.md

```diff
@@ -30,6 +30,7 @@
 * [BUGFIX] Experimental TSDB: fixed `/all_user_stats` and `/api/prom/user_stats` endpoints when using the experimental TSDB blocks storage. #2042
 * [BUGFIX] Experimental TSDB: fixed ruler to correctly work with the experimental TSDB blocks storage. #2101
 * [BUGFIX] Azure Blob ChunkStore: Fixed issue causing `invalid chunk checksum` errors. #2074
+* [BUGFIX] Experimental Memberlist KV store can now be used in single-binary Cortex. Attempts to use it previously would fail with panic. This change also breaks existing binary protocol used to exchange gossip messages, so this version will not be able to understand gossiped Ring when used in combination with the previous version of Cortex. #2016
```

Comment:
> This change also breaks existing binary protocol used to exchange gossip messages, so this version will not be able to understand gossiped Ring when used in combination with the previous version of Cortex.

Because of this may worth switching from `[BUGFIX]` to `[CHANGE]`. Could you also explain practically how a Cortex user using memberlist should upgrade to the next Cortex release? If it's more than a sentence, we may add a sub section after the changelog entries in `master` like we did for other changes in `0.6.0`.

Reply:
Moved it to CHANGE. Ideally to upgrade one would shutdown old Cortex, and start everything with new version... I know that we didn't do that :) We simply rolled out new version to the cluster, and since distributors are typically fastest ones to restart, they couldn't dispatch writes until new ingesters joined. But after they did, everything worked fine again.

I don't think that anybody else really runs this setup at the moment.